src/apps/js_v8/tmpl/request_authn_identity.cpp

```diff
@@ -0,0 +1,408 @@
+      return RequestEmptyAuthnIdentity::wrap(context, *empty_ident);
+    if (auto jwt_ident = endpoint_ctx.try_get_caller<ccf::JwtAuthnIdentity>())
+      return RequestJwtAuthnIdentity::wrap(context, *jwt_ident);
+
+    ReadOnlyTx& tx = endpoint_ctx.tx;
+    if (
+      auto user_cert_ident =
+        endpoint_ctx.try_get_caller<ccf::UserCertAuthnIdentity>())
+      return RequestUserCertAuthnIdentity::wrap(
+        context, *user_cert_ident, endpoint_registry, tx);
+    if (
+      auto member_cert_ident =
+        endpoint_ctx.try_get_caller<ccf::MemberCertAuthnIdentity>())
+      return RequestMemberCertAuthnIdentity::wrap(
+        context, *member_cert_ident, endpoint_registry, tx);
+    if (
+      auto user_sig_ident =
+        endpoint_ctx.try_get_caller<ccf::UserSignatureAuthnIdentity>())
+      return RequestUserSignatureAuthnIdentity::wrap(
+        context, *user_sig_ident, endpoint_registry, tx);
+    if (
+      auto member_sig_ident =
+        endpoint_ctx.try_get_caller<ccf::MemberSignatureAuthnIdentity>())
+      return RequestMemberSignatureAuthnIdentity::wrap(
+        context, *member_sig_ident, endpoint_registry, tx);
+    LOG_FATAL_FMT("Unknown caller type");
+    return v8::Null(isolate);
+  }
+
+  template <typename T>
+  void set_policy_name(v8::Isolate* isolate, v8::Local<v8::ObjectTemplate> tmpl)
+  {
+    tmpl->Set(
+      isolate,
+      "policy",
+      v8_util::to_v8_istr(isolate, ccfapp::get_policy_name_from_ident<T>()));
+  }
+
+  v8::Local<v8::ObjectTemplate> RequestEmptyAuthnIdentity::create_template(
+    v8::Isolate* isolate)
+  {
+    v8::EscapableHandleScope handle_scope(isolate);
+
+    v8::Local<v8::ObjectTemplate> tmpl = v8::ObjectTemplate::New(isolate);
+
+    tmpl->SetInternalFieldCount(0);
+
+    set_policy_name<ccf::EmptyAuthnIdentity>(isolate, tmpl);
+
+    return handle_scope.Escape(tmpl);
+  }
+
+  v8::Local<v8::Object> RequestEmptyAuthnIdentity::wrap(
+    v8::Local<v8::Context> context, const ccf::EmptyAuthnIdentity&)
+  {
+    v8::Isolate* isolate = context->GetIsolate();
+    v8::EscapableHandleScope handle_scope(isolate);
+
+    v8::Local<v8::ObjectTemplate> tmpl =
+      get_cached_object_template<RequestEmptyAuthnIdentity>(isolate);
+
+    v8::Local<v8::Object> result = tmpl->NewInstance(context).ToLocalChecked();
+
+    return handle_scope.Escape(result);
+  }
+
+  static ccf::JwtAuthnIdentity* unwrap_jwt_authn_identity(
+    v8::Local<v8::Object> obj)
+  {
+    return static_cast<ccf::JwtAuthnIdentity*>(
+      obj->GetAlignedPointerFromInternalField(0));
+  }
+
+  static void get_jwt_object(
+    v8::Local<v8::Name> name, const v8::PropertyCallbackInfo<v8::Value>& info)
+  {
+    ccf::JwtAuthnIdentity* jwt_ident = unwrap_jwt_authn_identity(info.Holder());
+    v8::Isolate* isolate = info.GetIsolate();
+    v8::Local<v8::Context> context = isolate->GetCurrentContext();
+
+    // Ideally, this should be a template as well.
```

Comment:
Is there a particular reason why it's not?

Reply:
No, I guess an extreme case of binding fatigue. I made it a template now.